### PR TITLE
Initializer macro for single UIO transfer

### PR DESCRIPTION
**Pull request by cahirwpz**
This is alternative solution to PR #274. I think I've found the cleanest solution to the problem, that does not require using rare C extensions.

### CODE REVIEW

**Comment by cahirwpz** [comment]
@coodie @rafalcieslak Could you compare this solution against the original?

File: include/uio.h [hunk 1]
```diff
@@ -21,6 +21,19 @@ typedef struct uio {
   vm_map_t *uio_vmspace; /* destination address space */
 } uio_t;
 
+#define UIO_SINGLE(op, vm_map, offset, buf, buflen)                            \
+  (uio_t) {                                                                    \
+    .uio_iov = (iovec_t[1]){(iovec_t){(buf), (buflen)}}, .uio_iovcnt = 1,      \
```
Comment (coodie) [inline include/uio.h]: Can you briefly explain how does that work? I don't see where you get space for iov.

Comment (cahirwpz) [inline include/uio.h]: The space for `iov` gets allocated on stack. With C99 (checked with `-std=c99 -Wpedantic`) you can do following:
```c
void foobar() {
  char *str;
  str = "foobar";
  str = (char[6]){'h', 'e', 'l', 'l', 'o', 0};
  int *tab;
  tab = (int[4]){1,2,3,4};
}
```
In both cases arrays (and string as well) will be allocated on stack (checked with optimizations turned off and `objdump -d`), unless they are constant (if so they're put into `.rodata` section). Does this explanation help?

File: tests/initrd.c [hunk 2]
```diff
@@ -12,19 +12,9 @@ static void dump_file(const char *path) {
...
-  uio.uio_iovcnt = 1;
-  uio.uio_vmspace = get_kernel_vm_map();
-  uio.uio_iov = &iov;
-  uio.uio_offset = 0;
-  iov.iov_base = buffer;
-  iov.iov_len = sizeof(buffer);
-  uio.uio_resid = sizeof(buffer);
 
+  uio_t uio;
+  uio = UIO_SINGLE_KERNEL(UIO_READ, 0, buffer, sizeof(buffer));
```
Comment (rafalcieslak) [inline tests/initrd.c]: Maybe:
```c
uio_t uio = UIO_SINGLE_KERNEL(UIO_READ, 0, buffer, sizeof(buffer));
```

**Pull request merged**